samples-android/ButtonClicker/jni/ButtonClickerNativeActivity_Engine.cpp

```diff
@@ -61,10 +60,12 @@ void Engine::UnloadResources() { renderer_.Unload(); }
  */
 int Engine::InitDisplay(const int32_t cmd) {
   if (!initialized_resources_) {
+    startup_mutex_.lock();
     gl_context_->Init(app_->window);
     InitUI();
     LoadResources();
     initialized_resources_ = true;
+    startup_mutex_.unlock();
```

Comment:
this is good point.  if there is exception in native code, it crashes and there is not any catcher here. I think java code handles lot of exceptions; in side native code, I do not see too much yet.